lib/simple_token_authentication.ex

```diff
@@ -12,19 +12,19 @@ defmodule SimpleTokenAuthentication do
 
   def call(conn, _opts) do
     val = get_auth_header(conn)
-    
-    tokens =
-      :simple_token_authentication
-      |> Application.get_env(:token)
-      |> List.wrap()
-      
-    service_tokens =
-      :simple_token_authentication
-      |> Application.get_env(:service_tokens)
-      |> List.wrap()
 
-    if Enum.any?(tokens ++ service_tokens, &matches?(&1, val)) do
-      conn
+    token_map =
+      case :persistent_term.get(:simple_token_authentication_map, nil) do
+        nil ->
+          build_token_map()
+
+        val ->
+          val
+      end
+
+    if service_name = token_map[val] do
```

Comment:
Do you know how vulnerable map lookups would be to timing attacks? It may not be an issue, but that would be my main concern with this change.

Reply:
I have the same question but since it is hashing and doing a lookup. I can only think there is a timing depending on the implementation of the hash (which I don t know the erlang one) and also how that affect string type keys

Reply:
So I think map keys are only hashed if there are at least 32 keys in the map, which I would imagine won't be the case for most apps using this.